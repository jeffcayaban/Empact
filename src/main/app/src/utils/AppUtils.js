import {API_BASE_URL, AUTHORIZATION_HEADER_FIELD} from '../constants';
import {ACCESS_TOKEN, AUTHORIZATION_TOKEN_TYPE, PETITION_LIST_SIZE} from "../constants";

/** Creates a request object and sends the request to the server. **/
const request = (options) => {
    const headers = new Headers({ 'Content-Type': 'application/json' });
    const accessToken = localStorage.getItem(ACCESS_TOKEN);

    if (accessToken !== undefined && accessToken !== null) {
        headers.append(AUTHORIZATION_HEADER_FIELD, `${AUTHORIZATION_TOKEN_TYPE} ${accessToken}`)
    }

    options = {...options, headers: headers};

    return fetch(options.url, options).then(response =>
            (response.json()).then(jsonResponse => (!response.ok) ? Promise.reject(jsonResponse) : jsonResponse)
    );
};

// ---------------------------- APIs ----------------------------

/** Gets all the petitions **/
export function getAllPetitions(page, size, sort, showClosed) {
    page = page || 0;
    size = size || PETITION_LIST_SIZE;

    return request({
        url: API_BASE_URL + "/petitions?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Gets a single petition given an ID **/
export function getPetition(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/" + petitionId,
        method: 'GET'
    });
}

/** Creates a new petition given its details **/
export function createPetition(petitionData) {
    return request({
        url: API_BASE_URL + "/petitions/create",
        method: 'POST',
        body: JSON.stringify(petitionData)
    });
}

/** Deletes an existing petition given its details **/
export function deletePetition(deletePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/delete",
        method: 'DELETE',
        body: JSON.stringify(deletePetitionData)
    });
}

/** Updates an existing petition with the given details **/
export function editPetition(editPetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/edit/",
        method: 'POST',
        body: JSON.stringify(editPetitionData)
    });
}

/** Logs in into a user account **/
export function login(loginRequest) {
    return request({
        url: API_BASE_URL + "/auth/signInUser",
        method: 'POST',
        body: JSON.stringify(loginRequest)
    });
}

/** Creates a new user account given the account details **/
export function signup(signupRequest) {
    return request({
        url: API_BASE_URL + "/auth/signUpUser",
        method: 'POST',
        body: JSON.stringify(signupRequest)
    });
}

/** Checks the username's availability given a username **/
export function checkUsernameAvailability(username) {
    return request({
        url: API_BASE_URL + "/user/checkUsernameAvailability?username=" + username,
        method: 'GET'
    });
}

/** Deletes a user **/
export function deleteUser(deleteUserRequest) {
    return request({
        url: API_BASE_URL + "/auth/deleteUser",
        method: 'DELETE',
        body: JSON.stringify(deleteUserRequest)
    })
}

/** Fetches the current user **/
export function getCurrentUser() {
    if(!localStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }

    return request({
        url: API_BASE_URL + "/user/getCurrentUser",
        method: 'GET'
    });
}

/** Fetches a user profile given a user's username **/
export function getUserProfile(username) {
    return request({
        url: API_BASE_URL + "/user/getUserProfile?username=" + username,
        method: 'GET'
    });
}

/** Fetches the petitions for a given user **/
export function getUserPetitions(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/user?username=" + username + "&page=" + page + "&size=" + size + "&sort=" + sort
        + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Fetches the most discussed petitions for a given user **/
export function getMostDiscussedPetitionsByUser(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitionsByUser?username=" + username + "&page=" + page +
            "&size=" + size + "&sort=" + sort
            + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Closes a petition given its details **/
export function closePetition(closePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/close",
        method: 'POST',
        body: JSON.stringify(closePetitionData)
    })
}

/** Creates a new argument by expert opinion given the new details **/
export function createArgumentByExpertOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Creates a new argument by popular opinion given the new details **/
export function createArgumentByPopularOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Fetches the arguments for a given petition **/
export function getArgumentsByRootPetitionId(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByRootPetitionId?rootPetitionId=" + petitionId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the arguments for a given argument **/
export function getArgumentsByArgumentId(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByArgumentId?argumentId=" + argumentId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the argument for a given ID **/
export function getArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/" + argumentId,
        method: 'GET'
    });
}

/** Fetches the parent argument for a given argument **/
export function getParentArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/getParentArgumentById?argumentId=" + argumentId,
        method: 'GET',
    });
}

/** Checks whether a petition is closed or not **/
export function isPetitionClosed(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/isPetitionClosed/" + petitionId,
        method: 'GET'
    })
}

/** Fetches the arguments for a given user **/
export function getUserArguments(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByUsernameAndAnonymity?username=" + username +
            "&showAnonymousContent=" + showAnonymousContent + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET'
    })
}

/** Fetches the number of supporting or opposing arguments for a given petition **/
export function countArgsByPetitionIdAndIsSupporting(petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countArgsByPetitionIdAndIsSupporting?petitionId=" + petitionId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Fetches the number of supporting or opposing arguments for a given argument **/
export function countArgsByArgumentIdAndIsSupporting(argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countAllByArgumentIdAndIsSupporting?argumentId=" + argumentId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Deletes an argument given its details **/
export function deleteArgument(deleteArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/delete",
        method: 'DELETE',
        body: JSON.stringify(deleteArgumentData)
    });
}

/** Updates an existing argument by popular opinion with new details **/
export function editArgumentByPopularOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Updates an existing argument by expert opinion with new details **/
export function editArgumentByExpertOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Fetches the most discussed petitions **/
export function getMostDiscussedPetitions(page, size, sort, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitions?page=" + page + "&size=" + size + "&sort=" + sort +
            "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments **/
export function getMostDiscussedArguments(pageSize, showClosed) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArguments?size=" + pageSize + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given petition **/
export function getMostDiscussedArgumentsForPetition(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForPetition?page=" + page + "&size=" + size + "&sort=" + sort +
            "&petitionId=" + petitionId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given argument **/
export function getMostDiscussedArgumentsForArgument(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForArgument?page=" + page + "&size=" + size + "&sort=" + sort +
            "&argumentId=" + argumentId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given user **/
export function getMostDiscussedArgumentsForUser(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForUser?page=" + page + "&size=" + size + "&sort=" + sort +
            "&username=" + username + "&showAnonymousContent=" + showAnonymousContent,
        method: 'GET'
    });
}

/** Fetches the data nodes required for the data visualisation of arguments **/
export function getDataNodes(contentId, isPetition) {
    return request({
        url: API_BASE_URL + "/arguments/getDataNodes?contentId=" + contentId + "&isPetition=" + isPetition,
        method: 'GET'
    });
}

/** Fetches data required for the petition report **/
export function getPetitionReport(petitionId) {
    return request({
        url: API_BASE_URL + "/arguments/getPetitionReport?petitionId=" + petitionId,
        method: 'GET'
    });
}

/** Searches for petitions given a query **/
export function searchPetitions(page, size, sort, showClosed, query) {
    return request({
        url: API_BASE_URL + "/petitions/search?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed
         + "&query=" + query,
        method: 'GET'
    });
}
