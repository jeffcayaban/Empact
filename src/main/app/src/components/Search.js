import React from 'react'

import {Button, FormGroup, FormControl, Grid, Col, Glyphicon, Checkbox} from 'react-bootstrap';
import Loader from "./Loader";
import {searchPetitions} from "../utils/AppUtils";
import PetitionsContentTable from "./common/PetitionsContentTable";
import {
    DISPLAY_SORT_BY_VALUES,
    NEWEST,
    SORT_BY_VALUES
} from "../constants";
import TableSortByBtn from "./common/TableSortByBtn";

/**
 * Is the search page that will allow the user to search for petitions.
 */

class Search extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            isLoading: false,
            query: "",

            petitions: null,
            page: 0,
            size: 5,
            totalNoOfResults: 0,
            totalPages: 0,
            last: false,

            sort: NEWEST,
            showClosed: false,
            currentSortByDisplay: NEWEST,
        };

        this.fetchPetitions = this.fetchPetitions.bind(this);
    }

    // Updates the component's state with the user's updated query.
    updateQuery(event) {
        const newQuery = event.target.value;
        if (this.state.query !== newQuery) {
            // Assumes that the query has changed.
            this.setState({ query: newQuery });
        }
    }

    // Updates the component's state to show or not show closed petitions.
    updateOnClosed() {
        this.setState({ showClosed: !this.state.showClosed });
    }

    // Updates the component's state to reflect the selected sort by option.
    handleChangeSortBy(key) {
        const { currentSortByDisplay } = this.state;

        const newSortByValue = SORT_BY_VALUES[key - 1];
        const newCurrentSortByDisplay = DISPLAY_SORT_BY_VALUES[key - 1];

        if (newCurrentSortByDisplay !== currentSortByDisplay) {
            // Assumes that the sort by option has changed.
            this.setState({ currentSortByDisplay: newCurrentSortByDisplay, sort: newSortByValue });
            this.fetchPetitions(0, newSortByValue);
        }
    }

    // Fetches the petitions that match the given query.
    fetchPetitions(page = this.state.page, sort = this.state.sort) {
        const { size, showClosed, query } = this.state;

        searchPetitions(page, size, sort, showClosed, query)
            .then(response => {
                this.setState({
                    isLoading: false,

                    petitions: response.content,
                    page: response.page,
                    size: response.size,
                    totalNoOfResults: response.totalElements,
                    totalPages: response.totalPages,
                    last: response.last
                })
            })
            .catch(error => this.setState({ isLoading: false }))
    }

    // Handles the request to fetch the petitions and displays a loader.
    handleSubmit(e) {
        e.preventDefault();
        this.setState({ isLoading: true });
        this.fetchPetitions();
    }

    // Generates the table of petitions given a set of petitions.
    generatePetitionsTable() {
        const { petitions, page, totalNoOfResults } = this.state;

        return (
            <Grid className={"divNoPadding"} style={{ paddingTop: '60px' }}>
                <Col md={12} lg={12}>
                    <div style={{ marginBottom: '15px' }}>
                        <TableSortByBtn
                            title={this.state.currentSortByDisplay}
                            onSelect={this.handleChangeSortBy.bind(this)}
                            totalContentCount={totalNoOfResults}
                            showMoreOptions={false}
                        />
                    </div>
                    <PetitionsContentTable
                        petitions={petitions}
                        pageNumber={page}
                        totalPages={totalNoOfResults}
                        showFollowUp={false}
                        loadFromServer={this.fetchPetitions}
                    />
                </Col>
            </Grid>
        )
    }

    render() {
        const { isLoading, query, petitions, showClosed } = this.state;

        if (!isLoading) {
            // Assumes that the data fetching is completed.

            return (
                <div style={{ paddingBottom: '30px' }}>
                    <div className={"centerStyle"}>
                        <h2>Search</h2>
                        <p>Search petitions on the site!</p>
                    </div>
                    <div className={"divNoPadding"} style={{ paddingTop: '40px' }}>
                        <form onSubmit={this.handleSubmit.bind(this)}>
                            <Grid>
                                <FormGroup bsSize="lg">
                                    <FormControl
                                        value={query}
                                        onChange={this.updateQuery.bind(this)}
                                        type="text"
                                        placeholder="Search"
                                    />
                                </FormGroup>
                            </Grid>
                            <Grid className={"showClosedCheckbox"}>
                                <Checkbox
                                    checked={showClosed}
                                    onChange={() => this.updateOnClosed()}
                                >
                                    Show Closed
                                </Checkbox>
                            </Grid>
                            <Grid>
                                <div className={"centerStyle"} style={{ paddingTop: '15px' }}>
                                    <Button type="submit" bsSize="large">
                                        Search
                                        <Glyphicon style={{ paddingLeft: '12px' }} glyph="search" />
                                    </Button>
                                </div>
                            </Grid>
                        </form>
                    </div>
                    { (petitions !== null) && this.generatePetitionsTable() }
                </div>
            );
        } else {
            // Assumes that the data is being fetched, so a loader is displayed.
            return <Loader isLoading={this.state.isLoading} />
        }
    }

}

export default Search;