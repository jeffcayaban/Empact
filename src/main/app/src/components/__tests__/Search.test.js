import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Search from "../Search";

import * as AppUtils from '../../utils/AppUtils';
import MockPetitions from '../../other/mock-data/MockGetPetitionsResponse';

describe('Search', () => {
    it('should render successfully', () => {
        const component = shallow(<Search />);
        component.setState({
            petitions: MockPetitions.content,
            page: MockPetitions.page,
            totalNoOfResults: MockPetitions.totalElements
        });

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render a loader', () => {
        const component = shallow(<Search />);
        component.setState({ isLoading: true });
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should successfully call fetchPetitions', () => {
        const originalSearchPetitions = AppUtils.searchPetitions;
        AppUtils.searchPetitions = () => Promise.resolve(MockPetitions);

        const component = shallow(<Search />);
        const instance = component.instance();

        instance.fetchPetitions(1, "asc");

        expect(toJson(component)).toMatchSnapshot();
        AppUtils.searchPetitions = originalSearchPetitions;
    });

    it('should successfully handle fetchPetitions failing', () => {
        const originalSearchPetitions = AppUtils.searchPetitions;
        AppUtils.searchPetitions = () => Promise.reject();

        const component = shallow(<Search />);
        const instance = component.instance();

        instance.fetchPetitions(1, "asc");

        expect(toJson(component)).toMatchSnapshot();
        AppUtils.searchPetitions = originalSearchPetitions;
    });

    it('should successfully call handleChangeSortBy', () => {
        const component = shallow(<Search />);
        const instance = component.instance();
        instance.handleChangeSortBy("2");
        expect(toJson(component)).toMatchSnapshot();
    });

    it('should successfully call handleSubmit', () => {
        const originalSearchPetitions = AppUtils.searchPetitions;
        AppUtils.searchPetitions = () => Promise.reject();

        const component = shallow(<Search />);
        const instance = component.instance();

        instance.handleSubmit({ preventDefault: jest.fn() });

        expect(toJson(component)).toMatchSnapshot();
        AppUtils.searchPetitions = originalSearchPetitions;
    });

    it('should successfully call updateQuery', () => {
        const component = shallow(<Search />);
        const instance = component.instance();
        instance.updateQuery({ target: { value: 'testValue' } });
        expect(component.state('query')).toBe('testValue');
    });

    it('should successfully call updateOnClosed', () => {
        const component = shallow(<Search />);
        const oldShowClosedValue = component.state('showClosed');
        const instance = component.instance();
        instance.updateOnClosed();
        expect(component.state('showClosed')).not.toBe(oldShowClosedValue);
    });
});