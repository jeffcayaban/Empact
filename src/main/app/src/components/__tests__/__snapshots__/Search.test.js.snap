// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render a loader 1`] = `
<Loader
  isLoading={true}
/>
`;

exports[`Search should render successfully 1`] = `
<div
  style={
    Object {
      "paddingBottom": "30px",
    }
  }
>
  <div
    className="centerStyle"
  >
    <h2>
      Search
    </h2>
    <p>
      Search petitions on the site!
    </p>
  </div>
  <div
    className="divNoPadding"
    style={
      Object {
        "paddingTop": "40px",
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <FormGroup
          bsClass="form-group"
          bsSize="lg"
        >
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </FormGroup>
      </Grid>
      <Grid
        bsClass="container"
        className="showClosedCheckbox"
        componentClass="div"
        fluid={false}
      >
        <Checkbox
          bsClass="checkbox"
          checked={false}
          disabled={false}
          inline={false}
          onChange={[Function]}
          title=""
        >
          Show Closed
        </Checkbox>
      </Grid>
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <div
          className="centerStyle"
          style={
            Object {
              "paddingTop": "15px",
            }
          }
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsSize="large"
            bsStyle="default"
            disabled={false}
            type="submit"
          >
            Search
            <Glyphicon
              bsClass="glyphicon"
              glyph="search"
              style={
                Object {
                  "paddingLeft": "12px",
                }
              }
            />
          </Button>
        </div>
      </Grid>
    </form>
  </div>
  <Grid
    bsClass="container"
    className="divNoPadding"
    componentClass="div"
    fluid={false}
    style={
      Object {
        "paddingTop": "60px",
      }
    }
  >
    <Col
      bsClass="col"
      componentClass="div"
      lg={12}
      md={12}
    >
      <div
        style={
          Object {
            "marginBottom": "15px",
          }
        }
      >
        <TableSortByBtn
          onSelect={[Function]}
          showMoreOptions={false}
          title="Newest"
          totalContentCount={1}
        />
      </div>
      <PetitionsContentTable
        loadFromServer={[Function]}
        pageNumber={0}
        petitions={
          Array [
            Object {
              "action": "TestAction",
              "closingDateTime": "2019-02-15T13:04:27Z",
              "createdBy": Object {
                "authorities": Array [
                  "ROLE_ADMIN",
                ],
                "firstName": "Admin",
                "id": 1,
                "lastName": "Account",
                "username": "admin",
              },
              "creationDateTime": "2019-02-14T13:04:29Z",
              "goal": "TestGoal",
              "id": 1,
              "isAnonymous": false,
              "isWinning": true,
              "lastUpdatedDateTime": "2019-02-14T13:04:38Z",
              "situation": "TestSituation",
              "title": "TestTitle",
              "value": "TestValue",
            },
          ]
        }
        showFollowUp={false}
        totalPages={1}
      />
    </Col>
  </Grid>
</div>
`;

exports[`Search should successfully call fetchPetitions 1`] = `
<div
  style={
    Object {
      "paddingBottom": "30px",
    }
  }
>
  <div
    className="centerStyle"
  >
    <h2>
      Search
    </h2>
    <p>
      Search petitions on the site!
    </p>
  </div>
  <div
    className="divNoPadding"
    style={
      Object {
        "paddingTop": "40px",
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <FormGroup
          bsClass="form-group"
          bsSize="lg"
        >
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </FormGroup>
      </Grid>
      <Grid
        bsClass="container"
        className="showClosedCheckbox"
        componentClass="div"
        fluid={false}
      >
        <Checkbox
          bsClass="checkbox"
          checked={false}
          disabled={false}
          inline={false}
          onChange={[Function]}
          title=""
        >
          Show Closed
        </Checkbox>
      </Grid>
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <div
          className="centerStyle"
          style={
            Object {
              "paddingTop": "15px",
            }
          }
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsSize="large"
            bsStyle="default"
            disabled={false}
            type="submit"
          >
            Search
            <Glyphicon
              bsClass="glyphicon"
              glyph="search"
              style={
                Object {
                  "paddingLeft": "12px",
                }
              }
            />
          </Button>
        </div>
      </Grid>
    </form>
  </div>
</div>
`;

exports[`Search should successfully call handleChangeSortBy 1`] = `
<div
  style={
    Object {
      "paddingBottom": "30px",
    }
  }
>
  <div
    className="centerStyle"
  >
    <h2>
      Search
    </h2>
    <p>
      Search petitions on the site!
    </p>
  </div>
  <div
    className="divNoPadding"
    style={
      Object {
        "paddingTop": "40px",
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <FormGroup
          bsClass="form-group"
          bsSize="lg"
        >
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </FormGroup>
      </Grid>
      <Grid
        bsClass="container"
        className="showClosedCheckbox"
        componentClass="div"
        fluid={false}
      >
        <Checkbox
          bsClass="checkbox"
          checked={false}
          disabled={false}
          inline={false}
          onChange={[Function]}
          title=""
        >
          Show Closed
        </Checkbox>
      </Grid>
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <div
          className="centerStyle"
          style={
            Object {
              "paddingTop": "15px",
            }
          }
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsSize="large"
            bsStyle="default"
            disabled={false}
            type="submit"
          >
            Search
            <Glyphicon
              bsClass="glyphicon"
              glyph="search"
              style={
                Object {
                  "paddingLeft": "12px",
                }
              }
            />
          </Button>
        </div>
      </Grid>
    </form>
  </div>
</div>
`;

exports[`Search should successfully call handleSubmit 1`] = `
<Loader
  isLoading={true}
/>
`;

exports[`Search should successfully handle fetchPetitions failing 1`] = `
<div
  style={
    Object {
      "paddingBottom": "30px",
    }
  }
>
  <div
    className="centerStyle"
  >
    <h2>
      Search
    </h2>
    <p>
      Search petitions on the site!
    </p>
  </div>
  <div
    className="divNoPadding"
    style={
      Object {
        "paddingTop": "40px",
      }
    }
  >
    <form
      onSubmit={[Function]}
    >
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <FormGroup
          bsClass="form-group"
          bsSize="lg"
        >
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Search"
            type="text"
            value=""
          />
        </FormGroup>
      </Grid>
      <Grid
        bsClass="container"
        className="showClosedCheckbox"
        componentClass="div"
        fluid={false}
      >
        <Checkbox
          bsClass="checkbox"
          checked={false}
          disabled={false}
          inline={false}
          onChange={[Function]}
          title=""
        >
          Show Closed
        </Checkbox>
      </Grid>
      <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <div
          className="centerStyle"
          style={
            Object {
              "paddingTop": "15px",
            }
          }
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsSize="large"
            bsStyle="default"
            disabled={false}
            type="submit"
          >
            Search
            <Glyphicon
              bsClass="glyphicon"
              glyph="search"
              style={
                Object {
                  "paddingLeft": "12px",
                }
              }
            />
          </Button>
        </div>
      </Grid>
    </form>
  </div>
</div>
`;
